src/steps/data_validation/utils/dataset_validator.py

```diff
@@ -0,0 +1,145 @@
+
+    Returns:
+        list[str]: A list containing the paths to all images found within the directory and its subdirectories.
+    """
+    image_path_list = []
+    for root, dirs, files in os.walk(image_dir):
+        for file in files:
+            image_path_list.append(os.path.join(root, file))
+    return image_path_list
+
+
+def validate_image_extraction(
+    dataset_context: DatasetContext, image_path_list: list[str]
+) -> None:
+    """
+    Validates that the number of extracted images matches the expected number of assets.
+
+    Args:
+        dataset_context (DatasetContext): The dataset context to validate.
+        image_path_list (list[str]): The list of image paths extracted from the dataset.
+
+    Raises:
+        ValueError: If the number of extracted images does not match the expected number of assets.
+    """
+    if len(image_path_list) < len(dataset_context.multi_asset):
+        raise ValueError(
+            f"Some images have not been extracted in {dataset_context.dataset_name} dataset"
+        )
+    if len(image_path_list) > len(dataset_context.multi_asset):
+        raise ValueError(
+            f"There are more images than expected in {dataset_context.dataset_name} dataset"
+        )
+
+
+def validate_image_corruption(
+    dataset_context: DatasetContext, image_path_list: list[str]
+) -> None:
+    """
+    Checks for corruption in the extracted images.
+
+    Parameters:
+        dataset_context (DatasetContext): The dataset context to validate.
+        image_path_list (list[str]): The list of image paths extracted from the dataset.
+
+    Raises:
+        ValueError: If any of the images are found to be corrupted.
+    """
+    for image_path in image_path_list:
+        try:
+            with Image.open(image_path) as img:
+                img.verify()  # Verify that this is a valid image
+        except Exception as e:
+            raise ValueError(
+                f"Image {image_path} is corrupted in {dataset_context.dataset_name} dataset - {e}"
+            )
+
+
+class DatasetValidator:
+    """
+    Validates various aspects of datasets within a dataset collection.
+
+    This class performs common validation tasks for datasets, including checking for image extraction
+    completeness, image format, image corruption, and annotation integrity. It serves as a base class
+    for more specific dataset validators.
+
+    Attributes:
+        dataset_collection (DatasetCollection): The collection of datasets to validate.
+    """
+
+    VALID_IMAGE_EXTENSIONS = (".jpg", ".jpeg", ".png")
+
+    def __init__(self, dataset_collection: DatasetCollection):
+        """
+        Initializes the DatasetValidator with a dataset collection to validate.
+
+        Parameters:
+            dataset_collection (DatasetCollection): The dataset collection to validate.
+        """
+        self.dataset_collection = dataset_collection
+
+    def validate_common(self) -> None:
```

Comment:
Since this is not the overwritten function (but `def validate()` instead), maybe this one can be prefixed with an underscore `_`?